### ISSUE TITLE
Built in filterBy methods - Add support for arguments as array

### ISSUE THREAD

**Issue by jenstornell** [getkirby/kirby#426]
**Working**

```php
$pages = $pages->filterBy( 'status', '!=',  'published' );
$pages = $pages->filterBy( 'status', '!=',  'private' );
```

**Don't work**

```php
$pages = $pages->filterBy( 'status', '!=', array( 'published', 'private' ) );
```

Because a value can not have two values at the same time, it should check if `status` in this case is `published` AND `private` in the case above. With `==` OR would be best.

For more advanced filters they can be custom made but this case I think could be just a natural evolution.

**Comment by lukasbestle** [comment]
Good idea, that's what I thought as well after I saw your post in the forum. However I think this should be a new filter like the following:

```php
$pages = $pages->filterBy( 'status', 'not in', array('published', 'private'));
```

Otherwise filtering by array values (like in models or in other types of collections) wouldn't work anymore.

**Comment by fvsch** [comment]
You can also implement your own collection filters:
https://getkirby.com/docs/advanced/filter-methods

But having built-in `'in'` and `'not in'` filters could be nice.
(I've had this need before, but used a more specific filtering callback instead, and lately changed it to a custom page method.)

Another possibility would be to change the implementation of existing filters, including `'=='` and `'!='` filters, to work with multiple values in an array.

For instance if the current implementation for `'=='` is:
```php
collection::$filters['=='] = function($collection, $field, $value) {
  foreach($collection->data as $key => $item) {
    if(collection::extractValue($item, $field) == $value) continue;
    unset($collection->$key);
  }
  return $collection;
};
```
it could be changed to:
```php
collection::$filters['=='] = function($collection, $field, $values) {
  if(is_string($values)) $values = [$values];
  foreach($collection->data as $key => $item) {
    if(in_array(collection::extractValue($item, $field), $values)) continue;
    unset($collection->$key);
  }
  return $collection;
};
```
but maybe that's overkill. (A bit too much jQuery-like, "take any input and work with it", mabye.)

**Comment by lukasbestle** [comment]
No, that's what I referred to above. If your collection has a method that returns an array, you won't be able to compare that array with an array you passed to the filter because Kirby would try to check for each value instead of just comparing the two arrays. That would break backwards-compatibility, so we need new filters.

**Comment by lukasbestle** [comment]
`in` and `not in` collection filters will be implemented in Kirby 2.3.2.
